replica/tablets.cc

```diff
@@ -461,10 +468,8 @@ future<std::optional<tablet_transition_stage>> read_tablet_transition_stage(cql3
 }
 
 stop_iteration tablet_sstable_set::for_each_sstable_set_until(const dht::partition_range& pr, std::function<stop_iteration(lw_shared_ptr<sstables::sstable_set>)> func) const {
-    size_t candidate_start = pr.start() ? group_of(pr.start()->value().token()) : size_t(0);
-    size_t candidate_end = pr.end() ? group_of(pr.end()->value().token()) : (_sstable_sets.size() - 1);
-    for (auto i = candidate_start; i <= candidate_end; i++) {
-        if (const auto& set = _sstable_sets[i]) {
+    for (const auto& i : subrange(pr)) {
+        if (const auto& set = _sstable_sets.at(i)) {
```

Comment:
else?  isn't it internal error?

Reply:
sg will never be nullptr here. we can stop using unique_ptr and this if will go away. we just need to switch to get_sstable_set(i) that finds i first and abort if not found.